### PR TITLE
using CefV8Value CreatePromise

### PR DESCRIPTION
**Pull request by osesantos**

### CODE REVIEW

File: CefGlue.BrowserProcess/ObjectBinding/JavascriptHelper.cs [hunk 1]
```diff
@@ -30,8 +28,7 @@ public static void Register(INativeObjectRegistry nativeObjectRegistry)
 
         public static PromiseHolder CreatePromise(this CefV8Context context)
         {
```
Comment (joaompneves) [inline CefGlue.BrowserProcess/ObjectBinding/JavascriptHelper.cs:30]: delete PromiseFactoryFunctionName and the js code

File: CefGlue.BrowserProcess/ObjectBinding/JavascriptHelper.cs [hunk 2]
```diff
@@ -30,8 +28,7 @@ public static void Register(INativeObjectRegistry nativeObjectRegistry)
...
-            var promiseFactory = GetGlobalInnerValue(context, PromiseFactoryFunctionName);
-            var promiseData = promiseFactory.ExecuteFunctionWithContext(context, null, new CefV8Value[0]); // create a promise and return the resolve and reject callbacks
+            var promiseData = CefV8Value.CreatePromise().ExecuteFunctionWithContext(context, null, new CefV8Value[0]); // create a promise and return the resolve and reject callbacks
 
             var promise = promiseData.GetValue("promise");
             var resolve = promiseData.GetValue("resolve");
```
Comment (joaompneves) [inline CefGlue.BrowserProcess/ObjectBinding/JavascriptHelper.cs:34]:
```suggestion
            var promise = CefV8Value.CreatePromise();
```

Comment (joaompneves) [reply CefGlue.BrowserProcess/ObjectBinding/JavascriptHelper.cs:34]: use the ResolvePromise and RejectPromise on the PromiseHolder

File: CefGlue.BrowserProcess/ObjectBinding/JavascriptHelper.cs (hunk 1 above)
Comment (osesantos) [reply CefGlue.BrowserProcess/ObjectBinding/JavascriptHelper.cs:30]: ok, thanks for the heads-up

File: CefGlue.BrowserProcess/ObjectBinding/JavascriptHelper.cs (hunk 2 above)
Comment (osesantos) [reply CefGlue.BrowserProcess/ObjectBinding/JavascriptHelper.cs]: done, I had to change the PromiseHolder Handlers to account for this new way

File: CefGlue.BrowserProcess/ObjectBinding/PromiseHolder.cs [hunk 3]
```diff
@@ -4,22 +4,17 @@ namespace Xilium.CefGlue.BrowserProcess.ObjectBinding
...
-        public delegate void RejectHandler(params CefV8Value[] args);
-
-        private readonly CefV8Value _resolve;
-        private readonly CefV8Value _reject;
+        public delegate void ResolveHandler(CefV8Value arg);
+        public delegate void RejectHandler(string errorMessage);
 
         public CefV8Value Promise { get; }
 
         public CefV8Context Context { get; }
```
Comment (joaompneves) [inline CefGlue.BrowserProcess/ObjectBinding/PromiseHolder.cs:12]: is this still being used?

File: CefGlue.BrowserProcess/ObjectBinding/PromiseHolder.cs [hunk 4]
```diff
@@ -4,22 +4,17 @@ namespace Xilium.CefGlue.BrowserProcess.ObjectBinding
...
     {
-        public delegate void ResolveHandler(params CefV8Value[] args);
-        public delegate void RejectHandler(params CefV8Value[] args);
-
-        private readonly CefV8Value _resolve;
-        private readonly CefV8Value _reject;
+        public delegate void ResolveHandler(CefV8Value arg);
+        public delegate void RejectHandler(string errorMessage);
 
         public CefV8Value Promise { get; }
```
Comment (joaompneves) [inline CefGlue.BrowserProcess/ObjectBinding/PromiseHolder.cs:10]: is this still being used?

Comment (osesantos) [reply CefGlue.BrowserProcess/ObjectBinding/PromiseHolder.cs:10]: I'm using the Promise internally on the PromiseHolder and there are some external references, like for instance, on the V8BuiltinFunctionHandler

**Pull request merged**